src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let joinedArray = '';
+
+    if (this[0] === null || this[0] === undefined) {
+      joinedArray = '';
+    } else {
+      joinedArray += this[0];
+    }
+
+    for (let i = 1; i < this.length; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
```
Let's do everything in one loop. Start the loop from the 0 index and check if it's not null and undefined. If so - add `this[i]`, also, add check if it's not the last element. If so - add `separator`